### DISCUSSION TITLE
Compatibility FDS and USFOS

### DISCUSSION THREAD

**Discussion by Ferdisss** [firemodels/fds#12510] [Q&A]
Hi,
Does anyone know if Fire Dynamics Simulator (FDS)can speak to USFOS (program for non-linear analysis of offshore structures), so I can use FDS to run fire response analyses from CFD in USFOS? Thank you!

**Comment by mcgratta** [comment]
FDS does not do structural analysis.

**Comment by mcgratta** [comment]
Are you asking if the results of FDS can be transferred to another program doing the structural analysis?

**Reply by Ferdisss** [reply]
Yes - if the data from FDS can be transferred to the structural program USFOS. Thank you!

**Comment by rmcdermo** [comment]
There are a few efforts to connect FDS with structural codes.  @RIO4FDS is the expert.  Have a look at this subfolder of our repository and explore it a bit to see if you find something that may be useful to you.

https://github.com/firemodels/fds/tree/master/Utilities/Structural_Interaction

**Comment by RIO4FDS** [comment]
Take a look at those folders, get the fds2fmti user's guide and let me know
if you have any questions.
https://drive.google.com/drive/folders/0BzJEIogPt2TyfnFERGNQNFVnWXJ4aXd5bFRhRzNDSktSN1o1dERFT3ZwaHJxNkM1T0ZMeXc?resourcekey=0-d5AdP45Gi4TWutPNUXz9rg

Regards,
Julio

**Comment by TristanHehnen** [comment]
Depending on what you want to do exactly, you could also set up your own interface. For example there is the `fdsreader` Python module that can access FDS results, e.g. `SLCF` or `BNDF`, and translates them into NumPy arrays. From there you could process them to your liking.

Here is an introduction:
https://firedynamics.github.io/LectureFireSimulation/content/tools/03_analysis/02_fdsreader.html

**Comment by Ferdisss** [comment]
Thank you! I ended up using FDS for finding heat flux on the surface, and putting this into fahts which works directly with USFOS.
